common/diagnostics/Metrics.java

```diff
@@ -546,15 +604,31 @@ JsonArray asJSON(String database, boolean isPrimaryServer) {
             return load;
         }
 
+        JsonArray asMonitoringJSON(String database, boolean isPrimaryServer) {
+            JsonArray load = new JsonArray();
+
+            if (isPrimaryServer && !isDeleted) {
+                JsonObject loadObject = new JsonObject();
+
+                loadObject.add("database", database);
+                loadObject.add("schema", schemaLoad.asJSON());
+                loadObject.add("data", dataLoad.asJSON());
+
+                load.add(loadObject);
+            }
+
+            return load;
+        }
+
         static String prometheusHeader() {
             return "# TYPE typedb_schema_data_count gauge";
         }
 
         String prometheusDiagnostics(String database, boolean isPrimaryServer) {
-            if (!isPrimaryServer || isDeleted) {
-                return "";
+            if (isPrimaryServer && !isDeleted) {
+                return schemaLoad.prometheusDiagnostics(database) + dataLoad.prometheusDiagnostics(database);
             }
-            return schemaLoad.prometheusDiagnostics(database) + dataLoad.prometheusDiagnostics(database);
+            return "";
         }
```

Comment:
Interesting decision to flip the condition here.

Before the change, this was essentially:
```java
if (!needsOutput)
    return ""; // bail
return output;
```
This became
```java
if (needsOutput)
    return output;
return "";
```

The former is a short-circuit followed by logic, the latter implies that the empty result is the default which is overridden by some condition.

At the very least the latter needs an `else` so the structure is obvious:
```java
if (needsOutput)
    return output;
else
    return "";
```

Although I would prefer if we stuck with the former.

Reply:
I've added `else` and decided to stick to the current version of the check to maintain the same check for two methods for better readability.